packages/odata-v2/test/test-util/test-data.ts

```diff
@@ -16,15 +16,24 @@ export const {
 export function createTestEntity(
   originalData: Record<string, any>
 ): TestEntity {
-  const entity = testEntityApi
-    .entityBuilder()
-    .keyPropertyGuid(originalData.KeyPropertyGuid)
-    .keyPropertyString(originalData.KeyPropertyString)
-    .stringProperty(originalData.StringProperty)
-    .booleanProperty(originalData.BooleanProperty)
-    .int16Property(originalData.Int16Property)
-    .build()
-    .setOrInitializeRemoteState();
+  const entityBuilder = testEntityApi.entityBuilder();
+
+  if ('KeyPropertyGuid' in originalData) {
```

Comment:
What was the motivation to create these `if (XY in YZ)` guards? Other than that lgtm 👍

Reply:
If one of the proprties is not defined in the data it will be set like : `{ property: undefined }` which is not what we want ;)

Reply:
Ah, I see, thanks for the explanation 👍 , so you basically polished the previous implementation while you were moving the function, makes sense.